block/submit.go

```diff
@@ -26,6 +27,17 @@ func (m *Manager) SubmitLoop(ctx context.Context) {
 
 	// defer func to clear the channels to release blocked goroutines on shutdown
 	defer func() {
+		// recover from panic, emit health status event and return the error
+		var err error
+		if r := recover(); r != nil {
+			m.logger.Error(fmt.Errorf("handle submission trigger: %v", r).Error())
+			err, _ = r.(error)
+		}
+		// new context because previous one is canceled
```

Comment:
the defer will return, it won't run the loop again

Reply:
yes